src/mds/MDLog.cc

```diff
@@ -977,8 +977,13 @@ void MDLog::_recovery_thread(MDSContext *completion)
     inodeno_t const default_log_ino = MDS_INO_LOG_OFFSET + mds->get_nodeid();
     jp.front = default_log_ino;
     int write_result = jp.save(mds->objecter);
-    // Nothing graceful we can do for this
-    ceph_assert(write_result >= 0);
+    if (write_result < 0) {
+      if (mds->is_daemon_stopping()) {
```

Comment:
We need to take the `mds_lock` before this ?

Reply:
You are right -- the lock ins't taken right at the stat but at places.

Reply:
okay, then the code should change from `mds->damaged_unlocked()` to `mds->damaged()`.

Reply:
Because `mds->damaged_unlocked()` already has a lock

Reply:
updated